widget/static/js/widget.js

```diff
@@ -0,0 +1,260 @@
+    'intro': {
+        'en-GB': 'of students were satisfied with their course.',
+        'cy-GB': ''
+    },
+    'ctaLead1': {
+        'en-GB': 'For ',
+        'cy-GB': ''
+    },
+    'ctaLead2': {
+        'en-GB': 'more',
+        'cy-GB': ''
+    },
+    'ctaLead3': {
+        'en-GB': ' official course information visit.',
+        'cy-GB': ''
+    },
+    'logo': {
+        'en-GB': 'Discover Uni',
+        'cy-GB': ''
+    },
+    'cta': {
+        'en-GB': 'See course data',
+        'cy-GB': ''
+    },
+    'noDataIntro': {
+        'en-GB': 'To see official information about this course and others visit Discover Uni.',
+        'cy-GB': ''
+    },
+    'noDataCtaLead1': {
+        'en-GB': 'Make an ',
+        'cy-GB': ''
+    },
+    'noDataCtaLead2': {
+        'en-GB': 'informed',
+        'cy-GB': ''
+    },
+    'noDataCtaLead3': {
+        'en-GB': ' choice.',
+        'cy-GB': ''
+    },
+    'noDataCta': {
+        'en-GB': 'See Discover Uni',
+        'cy-GB': ''
+    }
+}
+
+var MODES = {
+    'full-time': 1,
+    'part-time': 2,
+}
+
+var LANGUAGE_KEYS = {
+    'en-GB': 'english',
+    'cy-GB': 'welsh'
+}
+
+var  DiscoverUniWidget = function(targetDiv) {
+    this.targetDiv = targetDiv;
+    this.setup();
+}
+
+DiscoverUniWidget.prototype = {
+    setup: function() {
+        this.institution = this.targetDiv.dataset.institution;
+        this.course = this.targetDiv.dataset.course;
+        this.kismode = this.targetDiv.dataset.kismode;
+        this.orientation = this.targetDiv.dataset.orientation;
+        this.language = this.targetDiv.dataset.language;
+        this.languageKey = LANGUAGE_KEYS[this.language];
+        this.size = this.targetDiv.dataset.size;
+
+        this.targetDiv.classList.add(this.orientation);
+
+        this.addCss();
+        this.loadCourseData();
+    },
+
+    addCss: function() {
+        var fontNode = document.createElement('link');
+        fontNode.href = "http://fonts.googleapis.com/css?family=Montserrat";
+        fontNode.rel = "stylesheet";
+        fontNode.type = "text/css";
+        styling = "{% styles %}";
+        var stylingNode = document.createElement('style');
+        var stylingTextNode = document.createTextNode(styling);
+        stylingNode.appendChild(stylingTextNode);
+        var widgetScript = document.getElementById('unistats-widget-script');
+        widgetScript.parentNode.insertBefore(stylingNode, widgetScript);
+        widgetScript.parentNode.insertBefore(fontNode, widgetScript);
+    },
+
+    loadCourseData: function() {
+        var that = this;
+        var xhttp = new XMLHttpRequest();
+        xhttp.addEventListener("load", function() {
+            if (this.status === 200) {
+                that.renderDataWidget(JSON.parse(this.response).course);
+            } else {
+                that.renderWidget();
+            }
+        });
+        base_url = "{{api_domain}}/dataset/institutions/{{uni_id}}/courses/{{course_id}}/modes/{{mode}}";
```

Comment:
fyi - i think i will create a new endpoint on the dataset api; this will allow us to sort out workloads better in azure api management (such as rate limiting different endpoints).

Reply:
OK. Do you want me to hold off merging this until the API has been updated?